### PR TITLE
Make formatter KiCad-style compliant and simplify stackup patching

### PR DESCRIPTION
**Pull request by akhilles**
Previously, layout sync wasn't quite idempotent. It was only idempotent after KiCad loads + saves the layout file and fixes the formatting. Now, it's really idempotent. This also eliminates the need for complex change detection + conditional patching. The logic is now much simpler: always patch board config and then format.

---

> [!NOTE]
> **Medium Risk**
> Changes how `.kicad_*` files are rewritten during layout sync and introduces new formatting logic, so small formatting/ordering differences could create large diffs or expose parser/formatter edge cases despite added tests.
> 
> **Overview**
> Makes KiCad S-expression output deterministic and KiCad-style compliant by introducing a new `pcb_sexpr::formatter` (`prettify`/`format_tree`) and switching schematic/symbol generation to use it.
> 
> Simplifies `pcb layout` stackup/layer patching by removing approximate-diff/conditional updates and instead structurally mutating the parsed PCB S-expression, then always rewriting the board with canonical formatting (atomic buffered write).
> 
> Extends `pcb fmt` to optionally format a single explicit KiCad S-expression file path (with `--check`/`--diff` support via unified diffs), while keeping default/discovery behavior limited to `.zen` files; updates snapshots accordingly and adds `similar` as a dependency.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 6eeaaca1d9caa9ce30b2ab783c66df9688f90324. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: crates/pcb-layout/src/lib.rs [hunk 1]
```diff
@@ -940,189 +938,142 @@ fn patch_netclass_patterns(
...
-    let layers_sexpr = zen_stackup.generate_layers_sexpr(4);
-    let stackup_sexpr = zen_stackup.generate_stackup_sexpr();
+    apply_stackup_sections(&mut board, layers, stackup)?;
 
-    // Use surgical string replacement to avoid parsing issues with hex numbers
-    let mut updated_content = pcb_content;
-    updated_content = replace_section_in_pcb_content(&updated_content, "layers", &layers_sexpr)?;
-    updated_content = replace_section_in_pcb_content(&updated_content, "stackup", &stackup_sexpr)?;
+    let updated_content =
+        pcb_sexpr::formatter::format_tree(&board, pcb_sexpr::formatter::FormatMode::Normal);
```
[the remark this answers is not in the document]
Comment (LK) [reply crates/pcb-layout/src/lib.rs:957]: why does every llm think kicad s expressions support comments

[the remark this answers is not in the document]
Comment (akhilles) [reply crates/pcb-layout/src/lib.rs:957]: this aligns with KiCad behavior
